demo/TESTDCB2.MLC

```diff
@@ -0,0 +1,64 @@
+*********************************************************************
+* z390 - Mainframe assembler emulator and run-time engine
+* Copyright (C) 2021 z390 Assembler LLC
+*
+* This file is part of z390.
+*
+* z390 is free software; you can redistribute it and/or modify
+* it under the terms of the GNU General Public License as published by
+* the Free Software Foundation; either version 2 of the License, or
+* (at your option) any later version.
+* z390 is distributed in the hope that it will be useful,
+* but WITHOUT ANY WARRANTY; without even the implied warranty of
+* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+* GNU General Public License for more details.
+*
+* You should have received a copy of the GNU General Public License 
+* along with this program; if not, see https://www.gnu.org/licenses.
+
+* Author - Don Higgins                                              *
+* Date   - 06/04/2024                                               *
+*********************************************************************
+* Same as demo/TESTDCB1 but illustrates EODAD that returns to GET
+*********************************************************************
+*
+TESTDCB2 SUBENTRY
+         WTO   'TESTDCB2 Copy SYSUT1 ASC test file to SYSUT2 ASC text'
+         OPEN  MF=(E,OFILES)       Open all files
+         MVI   EOFFLAG,C'N'        Set not EOF on input file
+         GET   SYSUT1,RECORD       Get first record
+LOOP     DS    0H
+         CLI   EOFFLAG,C'Y'        No more records?
+         BE    ELOOP               Yes: all done
+         AP    PTOT,=P'1'          Count records
+         MVC   DTOT,=X'40202020'   Convert record count
+         ED    DTOT,PTOT           ... to printable decimal 
+         PUT   SYSOUT,MSG          Put rec# || record
+         PUT   SYSUT2,RECORD       Put record
+         GET   SYSUT1,RECORD       Next record
+         B     LOOP
+ELOOP    DS    0H
+         CLOSE MF=(E,CFILES)
+         WTO   'TESTDCB2 Ended OK'
+         SUBEXIT
+***********************************************************************
+*        EODAD routine
+***********************************************************************
+EOF      DS    0H
+         MVI   EOFFLAG,C'Y'        Set end of file reached on input
+         BR    14                  Return to GET
```

Comment:
Since the plan is to test that R14 points to after the GET instruction, consider a CL with an Acon for the expected address. Having two GET instructions makes it more complicated. Having only 1 GET (at the top of the loop) might make this a bit easier...